svm/src/transaction_processor.rs

```diff
@@ -726,7 +726,7 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
                 executed_units,
                 accounts_data_len_delta,
             },
-            programs_modified_by_tx: Box::new(programs_modified_by_tx),
+            programs_modified_by_tx: programs_modified_by_tx.entries().clone(),
```

Comment:
Why is the `.clone()` necessary here?

https://doc.rust-lang.org/std/collections/hash_map/struct.IntoIter.html

Reply:
I'm not sure what the link is suggesting, but it's to clone out of the reference. The alternative is to `std::mem::take`, which might work better here.

Reply:
You got it. Done!